pandas/tests/arrays/test_datetimes.py

```diff
@@ -89,11 +89,26 @@ def test_non_array_raises(self):
         with pytest.raises(ValueError, match="list"):
             DatetimeArray([1, 2, 3])
 
-    def test_other_type_raises(self):
+    def test_bool_dtype_raises(self):
+        arr = np.array([1, 2, 3], dtype="bool")
+
         with pytest.raises(
             ValueError, match="The dtype of 'values' is incorrect.*bool"
         ):
-            DatetimeArray(np.array([1, 2, 3], dtype="bool"))
+            DatetimeArray(arr)
```

Comment:
why is this a ValueError when DatetimeArray._from_sequence(arr) is a TypeError? Is is also worth considering making error messages consistent?

Reply:
originally i made it type error, then changed for consistenty with... something, but i dont remember what.  will try to figure that out and report bakc